microcosm_flask/conventions/config.py

```diff
@@ -0,0 +1,86 @@
+"""
+Config sharing convention.
+
+Reports service config protected by basic auth for securely running services
+locally with realistic config.
+
+"""
+from distutils.util import strtobool
+from json import dumps, loads
+
+from microcosm.api import defaults
+from microcosm.loaders.compose import PartitioningLoader
+from microcosm_flask.audit import skip_logging
+from microcosm_flask.conventions.base import Convention
+from microcosm_flask.conventions.encoding import make_response
+from microcosm_flask.namespaces import Namespace
+from microcosm_flask.operations import Operation
+
+
+class Config:
+    """
+    Wrapper around service config state.
+
+    """
+    def __init__(self, graph, include_build_info=True):
+        self.graph = graph
+        self.name = graph.metadata.name
+
+    def to_dict(self):
+        """
+        Encode the name, the status of all checks, and the current overall status.
+
+        """
+        if not isinstance(self.graph.loader, PartitioningLoader):
+            return dict(msg="Config sharing disabled for non-partioned loader")
+
+        if not hasattr(self.graph.loader, "secrets"):
+            return dict(msg="Config sharing disabled if no secrets are labelled")
+
+        def remove_nulls(dct):
+            return {key: value for key, value in dct.items() if value is not None}
+
+        return loads(
```

Comment:
I could do more work to include a sanitized version of secrets, e.g. 
```
{
config = ...,
secrets=...,
}
```
But as an unintended feature, the defaults for the secrets will still get picked up from the factories so it should be easy enough to edit in the valid values as needed.